programs/test/benchmark.c

```diff
@@ -648,7 +648,6 @@ int main( int argc, char *argv[] )
         TIME_AND_TSC( "HMAC_DRBG SHA-1 (NOPR)",
                 if( mbedtls_hmac_drbg_random( &hmac_drbg, buf, BUFSIZE ) != 0 )
                 mbedtls_exit(1) );
-        mbedtls_hmac_drbg_free( &hmac_drbg );
```

Comment:
I definitely think that calling mbedtls_hmac_drbg_free() here is not correct, but I am not sure whether just removing it should work in all cases. That is, there are probably no issues with just calling mbedtls_hmac_drbg_seed() repeatedly when using the default software code, but I am not sure what happens if, for example, the implementation is a hardware accelerator. Are we expected to call init->seed->seed->free or is it necessary to do init->seed->free->init->seed->free?

Also, I was trying to check what happens inside mbedtls_hmac_drbg_seed(), so I looked up the documentation for mbedtls_md_setup() which is called internally from seed and found the following:
```
 *                  It should be called after mbedtls_md_init() or
 *                  mbedtls_md_free(). Makes it necessary to call
 *                  mbedtls_md_free() later.
```
So according to this, it not necessary to call init at all when using the MD abstraction?

Reply:
@andresag01 if by HW acceleration you mean the HW entropy source, then I don't see a problem. Calling seed the second time is *similar* to reseeding. The HW should be stateless and should return the desirted entropy whenever asked.

Reply:
Ok